artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -976,17 +927,97 @@ public RoutingStatus route(final Message message,
       return result;
    }
 
-   // HORNETQ-1029
-   private void applyExpiryDelay(Message message, SimpleString address) {
-      long expirationOverride = addressSettingsRepository.getMatch(address.toString()).getExpiryDelay();
+   private static void routeAndUpdateMessageCounter(final Message message,
+                                                    final RoutingContext context,
+                                                    final Binding bindingMove,
+                                                    final Bindings bindings,
+                                                    final SimpleString address,
+                                                    final AddressInfo addressInfo) throws Exception {
+      if (bindingMove != null) {
+         context.clear();
+         bindingMove.route(message, context);
+         if (addressInfo != null) {
+            addressInfo.incrementRoutedMessageCount();
+         }
+      } else if (bindings != null) {
+         context.clear();
```

Comment:
In the original version you did `context.clear` was present only before `bindingMove.route`: I have added it before `bindings.move` too: makes sense?

Reply:
actually no.. this will always clear the context, and there's no reused here.

I'm removing it here

Reply:
@franz1981  ^^ (just to notifcate you)